### DISCUSSION TITLE
Request headers in the FlareSolverr response dto

### DISCUSSION THREAD

**Discussion by kipy-be** [FlareSolverr/FlareSolverr#1213] [Ideas]
After a few testings with a dotnet app using FlareSolverr to retrieve cf_clearance and UserAgent, it's seems not to be enough for CF to accept later request with a httpclient.

I've been able to make a successful request by copying all the headers of the browser. (It seems like CF is checking all the "legit" headers a specific browser can send, and blocks the request if only one is missing).

That would be nice if FlareSolverr server could pass all the request headers in addition of the UserAgent and the cookie :) (there are the headers of the response, but not of the request)

**Comment by ewwink** [comment]
Too bad that Selenium doesn't natively support returning Request Headers. 
May we know why FlareSolverr still doesn't use Selenium Wire or the Browser DevTools Protocol?

```
from seleniumwire import webdriver

driver = webdriver.Chrome()
driver.get("https://www.example.com")
# Access the last request made
request = driver.requests[-1]
print(request.headers)
```


https://github.com/FlareSolverr/FlareSolverr/blob/c7505e3cbf6836385c1cb7f72e4807bc9b1acf36/src/flaresolverr_service.py#L392
